Rect42.dart

```diff
@@ -0,0 +1,121 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+
+class Rect42 extends StatefulWidget {
+  final int count;
+
+  const Rect42({Key? key, required this.count}) : super(key: key);
+
+  @override
+  State<Rect42> createState() => _Rect42State();
+}
+
+class ListColour<listRow> {
+  // class created to generate a radom colour for each row within the the ListView
+  late Color tileColour =
+      Colors.primaries[Random().nextInt(Colors.primaries.length)];
+
+  void changeColour1(tileColour) {
+    //method creatd to change the colour of the row
+    this.tileColour = tileColour;
+    tileColour = Colors.primaries[Random().nextInt(Colors.primaries.length)];
+  }
+}
+
+class _Rect42State extends State<Rect42> {
+  late List tiles;
+
+  @override
+  void initState() {
+    super.initState();
+    tiles = List.generate(widget.count, (index) => ListColour().tileColour);

```

Comment:
this should come from bloc or cubit

Reply:
I have moved this logic to a Bloc I created in the second pull request I uploaded.